param/parameterized.py

```diff
@@ -1072,16 +1072,43 @@ def __init__(self_, cls, self=None):
         """
         self_.cls = cls
         self_.self = self
-        self_._BATCH_WATCH = False  # If true, Event and watcher objects are queued.
-        self_._TRIGGER = False
-        self_._events = []         # Queue of batched eventd
-        self_._watchers = []         # Queue of batched watchers
+
+    @property
+    def _BATCH_WATCH(self_):
+        return self_.self_or_cls._parameters_state['BATCH_WATCH']
```

Comment:
To allow `.param` to be a property any state that was previously stored on `Parameters` now needs to be stored on the Parameterized itself. I didn't want to rock the boat too much so for now I've simply proxied the state that is now on the Parameterized class or instance using these properties. If we want to refactor more then all references to `parameterized.param._BATCH_WATCH`, `parameterized.param._TRIGGER`, `parameterized.param._events` and `parameterized.param._watchers` could be updated to refer to the `parameterized._parameters_state` instead.